### PR TITLE
indexes: Add new path to list all indexes

### PR DESCRIPTION
**Pull request by tloisNeo9**

### CODE REVIEW

**Review by BenjD90 — CHANGES_REQUESTED** [review]
- add missing tests
- fix type on function return

File: src/index-manager.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export class IndexManager {
 	/**
 	 * Returns a list of all indexes.
 	 */
-	public async findAllIndexes(): Promise<any> {
+	public async findAllIndexes(): Promise<CommandCursor> {
```
Comment (BenjD90) [inline src/index-manager.ts:16]: add a more precise type

File: test/indexes.test.ts [hunk 2]
```diff
@@ -21,6 +21,26 @@ ava('[Indexes] Create index', async (t: Assertions) => {
...
+	const collection = global.db.collection(`test-${Date.now()}`);
+	const mongoClient = new MongoClient(collection, Object, null);
+
+	await mongoClient.createIndex('index_1');
+	await mongoClient.createIndex('index_2');
+	await mongoClient.createIndex('index_3');
+
+	const createdIndexes = await (await mongoClient.findAllIndexes()).toArray();
+	const correctIndexes =
+		createdIndexes[1].name === 'index_1_1' &&
```
Comment (BenjD90) [inline test/indexes.test.ts:34]: could you sur assertions for that :  

t.is(createdIndexes[1].name, 'index_1_1', 'index 1 created and listed')
...

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/client.ts [hunk 3]
```diff
@@ -115,7 +116,7 @@ export class MongoClient<U extends BaseMongoObject, L extends BaseMongoObject> {
 		return new MongoClient<U, L>(newName, this.type, this.typeList, this.conf);
 	}
 
-	public async findAllIndexes(): Promise<CommandCursor> {
+	public async findAllIndexes(): Promise<IndexSpecification[]> {
 		return await this.indexManager.findAllIndexes();
```
Comment (BenjD90) [inline src/client.ts:120]: the toArray() is missing here

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/index-manager.ts [hunk 4]
```diff
@@ -10,6 +10,17 @@ export class IndexManager {
 	 */
 	constructor(private readonly collection: Collection) {}
 
+	/**
+	 * Returns a list of all indexes.
+	 */
+	public async findAllIndexes(): Promise<IndexSpecification[]> {
+		try {
+			return await this.collection.listIndexes().toArray();
```
Comment (BenjD90) [inline src/index-manager.ts:18]: :warning: confusion in promises 

```ts
await (await this.collection.listIndexes()).toArray()
```

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/index-manager.ts (hunk 4 above)
Comment (tloisNeo9) [reply src/index-manager.ts:18]: Is the inner await really necessary in this situation?

**Pull request merged**